addons/mrp/models/mrp_bom.py

```diff
@@ -127,6 +127,8 @@ def onchange_product_tmpl_id(self):
                 self.product_id = False
             for line in self.bom_line_ids:
                 line.bom_product_template_attribute_value_ids = False
+            if self.product_tmpl_id.bom_count:  # add a reference to the bom if there is already a bom for this product
+                self.code = _("%s (copy) %s" % (str(self.product_tmpl_id.name), self.product_tmpl_id.bom_count))
```

Comment:
It should probablybe a compute to work with copy. Also it's better to use display_name in this case